src/Ladybug_Shadow Study.py

```diff
@@ -426,6 +430,18 @@ def isShadedByFaces(self, surface):
             shaded.append(toBeShadedFace.boundary)
 
 
-        shadow = shadowCrvsCollection
-        shade = shaded
-
+        shadowCurves = shadowCrvsCollection
+        shadeCurves = shaded
+        
+        shadow = []
+        shade = []
+        
+        for item in shadowCurves:
+            shadow.append(rs.AddPlanarMesh(item))
```

Comment:
@devngc sorry that I didn't catch this one. `rhinoscriptsyntax` can be quite slow in cases such as this. You can use Rhinocommon to generate the meshes. Here is a sample code:

```python
import Rhino as rc
meshingPar = rc.Geometry.MeshingParameters.Coarse
mesh = rc.Geometry.Mesh.CreateFromPlanarBoundary(crv, meshingPar)
```